packages/yoga/src/IconWrapper/IconWrapper.jsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { oneOf, elementType, string } from 'prop-types';
+import { withTheme } from 'styled-components';
+
+const IconWrapper = ({
+  as: Component,
+  width,
+  height,
+  fill,
+  stroke,
+  theme,
+  ...props
+}) => (
+  <Component
+    width={theme.yoga.spacing[width] || width}
+    height={theme.yoga.spacing[height] || height}
+    {...(fill && { fill: theme.yoga.colors[fill] || fill })}
+    {...(stroke && { stroke: theme.yoga.colors[stroke] || stroke })}
+    {...props}
+  />
+);
+
+const commonSizes = [
+  'xxxsmall',
+  'xxsmall',
+  'xsmall',
+  'small',
+  'medium',
+  'large',
+  'xlarge',
+  'xxlarge',
+  'xxxlarge',
+  'huge',
+  'xhuge',
+];
+
+IconWrapper.propTypes = {
+  as: elementType.isRequired,
+  fill: string,
+  stroke: string,
+  width: oneOf(commonSizes),
+  height: oneOf(commonSizes),
```

Comment:
If we also wan't to be able to use "custom" values, we should reflect it at the `propTypes`, don't we?

`allColors || String` and `allSizes || Number`.

Reply:
Hmmmm, would this work?

```js
oneOf([...commonSizes, instanceOf(Number)])
```

Pretty sure it won't, but we can try. 🤣

Another option would be this (as @ggdaltoso [said](https://github.com/Gympass/yoga/pull/212#discussion_r573854431)):

```js
oneOfType([string, number])
```

Reply:
Actually, the first suggestion didn't triggered any console error, warning or crash. Nice to know, never thought of it 🤔